tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -250,6 +254,108 @@ static void test_nanocoap__get_path_too_long(void)
     TEST_ASSERT_EQUAL_INT(-ENOSPC, get_len);
 }
 
+/*
+ * Helper for server_get tests below.
+ * Request for nanocoap server example /riot/value resource.
+ * Includes 2-byte token; non-confirmable.
+ */
+static int _read_riot_value_req(coap_pkt_t *pkt, uint8_t *buf)
+{
+    uint8_t pkt_data[] = {
+        0x52, 0x01, 0x9e, 0x6b, 0x35, 0x61, 0xb4, 0x72,
+        0x69, 0x6f, 0x74, 0x05, 0x76, 0x61, 0x6c, 0x75,
+        0x65
+    };
+    memcpy(buf, pkt_data, sizeof(pkt_data));
+
+    return coap_parse(pkt, buf, sizeof(pkt_data));
+}
+
+/* Server GET request success case. */
+static void test_nanocoap__server_get_req(void)
+{
+    uint8_t buf[_BUF_SIZE];
+    coap_pkt_t pkt;
+    char path[] = "/riot/value";
+
+    int res = _read_riot_value_req(&pkt, &buf[0]);
+
+    TEST_ASSERT_EQUAL_INT(0, res);
+    TEST_ASSERT_EQUAL_INT(COAP_METHOD_GET, coap_get_code(&pkt));
+    TEST_ASSERT_EQUAL_INT(2, coap_get_token_len(&pkt));
+    TEST_ASSERT_EQUAL_INT(4 + 2, coap_get_total_hdr_len(&pkt));
+    TEST_ASSERT_EQUAL_INT(COAP_TYPE_NON, coap_get_type(&pkt));
+    TEST_ASSERT_EQUAL_INT(0, pkt.payload_len);
+
+    char uri[64] = {0};
+    coap_get_uri_path(&pkt, (uint8_t *)&uri[0]);
+    TEST_ASSERT_EQUAL_STRING((char *)path, (char *)uri);
+}
+
+/* Response for server GET request using coap_reply_simple(). */
+static void test_nanocoap__server_reply_simple(void)
+{
+    uint8_t buf[_BUF_SIZE];
+    coap_pkt_t pkt;
+    char payload[2] = "0";
+
+    int res = _read_riot_value_req(&pkt, &buf[0]);
+
+    coap_reply_simple(&pkt, COAP_CODE_CONTENT, buf, _BUF_SIZE,
+                      COAP_FORMAT_TEXT, (uint8_t *)payload, 1);
+
+    TEST_ASSERT_EQUAL_INT(0, res);
+    TEST_ASSERT_EQUAL_INT(COAP_CODE_CONTENT, coap_get_code_raw(&pkt));
+    TEST_ASSERT_EQUAL_INT(2, coap_get_token_len(&pkt));
+    TEST_ASSERT_EQUAL_INT(4 + 2, coap_get_total_hdr_len(&pkt));
+    TEST_ASSERT_EQUAL_INT(COAP_TYPE_NON, coap_get_type(&pkt));
+}
+
+/*
+ * Helper for server_get tests below.
+ * Request for nanocoap server example /riot/value resource.
+ * Includes 2-byte token; confirmable.
+ */
+static int _read_riot_value_req_con(coap_pkt_t *pkt, uint8_t *buf)
+{
+    uint8_t pkt_data[] = {
```

Comment:
same here, looks different from the other one, still unclear to me what and why

Reply:
Byte 1 is different since confirmable, and this is noted in the comment. Bytes 3 and 4 encode a different message ID, but I didn't think that was worth commenting on.